tracklytics-plugin/src/main/groovy/tracklytics/weaving/plugin/TracklyticsPlugin.groovy

```diff
@@ -12,42 +12,72 @@ class TracklyticsPlugin implements Plugin<Project> {
   void apply(Project project) {
 
     project.dependencies {
-      compile 'org.aspectj:aspectjrt:1.8.10'
-      compile 'com.orhanobut.tracklytics:tracklytics-runtime:2.0.0'
+      implementation 'org.aspectj:aspectjrt:1.8.10'
+      implementation 'com.orhanobut.tracklytics:tracklytics-runtime:2.1.0'
     }
 
     project.android.applicationVariants.all { variant ->
-      JavaCompile javaCompile = variant.javaCompile
-      String[] args = [
-          "-showWeaveInfo",
-          "-1.7",
-          "-inpath", javaCompile.destinationDir.toString(),
-          "-aspectpath", javaCompile.classpath.asPath,
-          "-d", javaCompile.destinationDir.toString(),
-          "-classpath", javaCompile.classpath.asPath,
-          "-bootclasspath", project.android.bootClasspath.join(File.pathSeparator)
-      ]
+      JavaCompile javaCompile
+      if (variant.hasProperty('javaCompileProvider')) {
+        // Android 3.3.0+
+        javaCompile = variant.javaCompileProvider.get()
+      } else {
+        javaCompile = variant.javaCompile
+      }
+
       // Gets the variant name and capitalize the first character
       def variantName = variant.name[0].toUpperCase() + variant.name[1..-1].toLowerCase()
 
       // Weave the binary for the actual code
       // CompileSources task is invoked after java and kotlin compilers and copy kotlin classes
       // That's the moment we have the finalized byte code and we can weave the aspects
       project.tasks.findByName('compile' + variantName + 'Sources')?.doLast {
+        def destinationDir = javaCompile.destinationDir.toString()
```

Comment:
I see that this particular code is duplicated. Did previous usage cause any issue?

Reply:
Yes,   _3. Moved String[] args to doLast task (because args can't be init before assembleTask)_ destinationDir is null before assembleTask

Reply:
So I moved it to `doLast`

Reply:
Thanks, makes sense 👍